relay-server/src/utils/rate_limits.rs

```diff
@@ -93,6 +93,7 @@ fn infer_event_category(item: &Item) -> Option<DataCategory> {
         ItemType::Event => Some(DataCategory::Error),
         ItemType::Transaction => Some(DataCategory::Transaction),
         ItemType::Security | ItemType::RawSecurity => Some(DataCategory::Security),
+        ItemType::Nel => None,
```

Comment:
`From<EventType> for DataCategory` returns `DataCategory::Error`, so we should probably also return the same here. That would mean NEL reports will be considered an error event for all intents and purposes, including rate limiting and billing. Considering that size limiting also counts NEL as events, that would be the most coherent option.